tests/security/test_comments_xss.py

```diff
@@ -0,0 +1,203 @@
+"""
+Test suite for comments XSS vulnerability fix.
+
+Tests that user comments are properly sanitized to prevent XSS attacks
+while preserving safe markdown formatting.
+"""
+import pytest
+from django.contrib.contenttypes.models import ContentType
+
+from vitrina.comments.models import Comment
+from vitrina.datasets.models import Dataset
+from vitrina.orgs.models import Organization
+from vitrina.users.models import User
+from vitrina.templatetags.markdown_tags import markdown
+
+
+# XSS payloads to test
+XSS_PAYLOADS = [
+    # Basic script injection
+    '<script>alert("XSS")</script>',
+    '<script>alert(String.fromCharCode(88,83,83))</script>',
+    
+    # Image onerror
+    '<img src=x onerror="alert(1)">',
+    '<img src=x onerror="window.location=\'http://attacker.com\'">',
+    
+    # SVG injection
+    '<svg onload="alert(1)">',
+    '<svg><script>alert(1)</script></svg>',
+    
+    # JavaScript URLs
+    '<a href="javascript:alert(1)">Click</a>',
+    '[Link](javascript:alert(1))',  # Markdown format
+    
+    # Event handlers
+    '<div onmouseover="alert(1)">Hover me</div>',
+    '<input onfocus="alert(1)" autofocus>',
+    '<body onload="alert(1)">',
+    
+    # Iframe injection
+    '<iframe src="javascript:alert(1)">',
+    '<iframe src="data:text/html,<script>alert(1)</script>">',
+]
+
+
+@pytest.fixture
+def user(db):
+    """Create test user."""
+    return User.objects.create_user(
+        email='test@example.com',
+        first_name='Test',
+        last_name='User',
+        password='testpass123'
+    )
+
+
+@pytest.fixture
+def organization(db):
+    """Create test organization."""
+    return Organization.add_root(
+        title='Test Organization',
+        company_code='123456'
+    )
+
+
+@pytest.fixture
+def dataset(db, organization):
+    """Create test dataset."""
+    return Dataset.objects.create(
+        title='Test Dataset',
+        organization=organization
+    )
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize("payload", XSS_PAYLOADS)
+def test_comment_blocks_xss_payload(user, dataset, payload):
+    """Test that comments block all XSS payloads."""
+    ct = ContentType.objects.get_for_model(dataset)
+    comment = Comment.objects.create(
+        user=user,
+        body=payload,
+        type=Comment.USER,
+        content_type=ct,
+        object_id=dataset.id
+    )
+    
+    # Get rendered HTML (simulating template rendering)
+    body = comment.body_text()
+    html = markdown(body)
+    html_lower = str(html).lower()
+    
+    # Dangerous content should be escaped (not executable)
+    # Check that we don't have UNESCAPED dangerous tags
+    assert '<script>' not in html_lower, f"Unescaped script tag found in: {html}"
```

Comment:
Would be nice to keep consistent with the quotation marks, either singles or doubles. E.g.:

```suggestion
    assert "<script>" not in html_lower, f"Unescaped script tag found in: {html}"
```

Obviously, not in places where you need a string that has double quotes inside it.